### DISCUSSION TITLE
Can Anyone Pip Install NiceGUI In Python 3.12?

### DISCUSSION THREAD

**Discussion by clueple** [zauberzeug/nicegui#1783] [Q&A]
### Question

I've installed Python 3.12.  But I couldn't pip install nicegui.
(It's too long and basically it's related to distutils being removed by Py 3.12)

The detail is listed in the Python official website:
https://peps.python.org/pep-0632/

I read that 'setuptools' could be the fix (I pip install setuptools and it works for flet, but not nicegui).

Wonder if anyone is successful installing nicegui in Python 3.12.

**Comment by falkoschindler** [comment]
We're currently waiting for https://github.com/MagicStack/uvloop/issues/567 before we can officially add support for Python 3.12.
For reference, [here](https://discord.com/channels/1089836369431498784/1158795388036325406/1158795388036325406) is a related discussion on Discord.

**Reply by falkoschindler** [reply]
Update: Right now we're waiting for aiohttp 3.9 to be released, which will add support for Python 3.12 (https://github.com/aio-libs/aiohttp/issues/7675).

**Reply by yelloball** [reply]
In the interim, pip installing 3.9.0b0 (beta) might work with Python 3.12
